src/network.py

```diff
@@ -120,25 +101,225 @@ def forward(self, enc_output):
         temp2 = self.dropout(self.non_linear(self.fc2(temp)))
         return self.fc3(temp2)
     
-class SpeechTransformer(SpeechNet):
+class SpeechTransformer(AutoEncoderNet):
     # TODO: Fill in with pre/post needed and enc/dec
-    def __init_(self):
+    def __init__(self, args):
         super(SpeechTransformer, self).__init__()
 
 
-class SpeechRNN(SpeechNet):
+class SpeechRNN(AutoEncoderNet):
     # TODO: Fill in with pre/post needed and enc/dec
-    def __init_(self):
+    def __init__(self, args):
         super(SpeechRNN, self).__init__()
+        self.prenet = SpeechPrenet(args.num_mels, args.hidden, args.e_in)
+        self.encoder = RNNEncoder(args.e_in, args.hidden, args.e_out, 
+            dropout=args.e_p, num_layers=args.e_layers, bidirectional=args.e_bi)
+        self.decoder = RNNDecoder(args.e_out, args.d_in, args.hidden, args.num_mels, 
+            dropout=args.d_p, num_layers=args.d_layers, attention=args.d_attn)
+        self.postnet = SpeechPostnet(args.num_mels, args.hidden)
+    
+
+    def preprocess(self, raw_input_tensor):
+        # raw_input_tensor should be a LongTensor of size [batch_size x seq_len x 1]
+        # should already be padded as well
+        # Get a mask of 0 for no padding, 1 for padding of size [batch_size x seq_len]
+        # NOTE: I do not think this is correct for mel... but check if all are padding in num_mels
+        pad_mask = torch.all(raw_input_tensor.eq(PAD_IDX), dim=-1).float().squeeze() 
+        return self.prenet(raw_input_tensor), pad_mask
```

Comment:
This is my understanding as well - this mask will effectively check all entries in the last dimension are PAD, and if that is the case then the time step was a padding timestep.  0's do not get used in anything else for text for sure, but for log_mel they are also used for the 0th timestep (like SOS)

Reply:
Interesting, is there a reason why they used it for the 0th timestep? I don't think it's too much of an issue to have a PAD at the start though.

Reply:
I think its because in using the mel filters, 0 represents silence, and any other value would be noise.  Its only used on the 0th timestep for the `mel_input` variable though, which is just the `mel` variable but with this 1 timestep of silence for conditioning on

Reply:
Okay, that consistency is nice!